rs/cli/src/cli.rs

```diff
@@ -423,3 +426,63 @@ pub mod nodes {
         },
     }
 }
+
+pub mod proposals {
+    use super::*;
+
+    #[derive(Parser, Clone)]
+    pub struct Cmd {
+        #[clap(subcommand)]
+        pub subcommand: Commands,
+    }
+
+    #[derive(Subcommand, Clone)]
+    pub enum Commands {
+        /// Get list of pending proposals
+        Pending,
+
+        /// Get list of filtered proposals
+        List {
+            /// Limit on the number of \[ProposalInfo\] to return. If no value is
+            /// specified, or if a value greater than 100 is specified, 100
+            /// will be used.
+            #[clap(long, default_value = "100")]
+            limit: u32,
+            /// If specified, only return proposals that are strictly earlier than
+            /// the specified proposal according to the proposal ID. If not
+            /// specified, start with the most recent proposal.
+            #[clap(long)]
+            before_proposal: Option<u64>,
+            /// Exclude proposals with a topic in this list. This is particularly
+            /// useful to exclude proposals on the topics TOPIC_EXCHANGE_RATE and
+            /// TOPIC_KYC which most users are not likely to be interested in
+            /// seeing.
+            #[clap(long)]
+            exclude_topic: Vec<i32>,
```

Comment:
This needs to be a list of strings, not a single string.  It should also be named in the plural to indicate it's a list.  Not sure how clap handles lists.

Reply:
Not the best solution right now. If you take a look at how `ListProposalInfo` struct is formatted it also expects a `Vec<i32>`. On top of that having strings means that we would need to keep those in sync and the basic rename would break while in reality it shouldn't. I would like to keep this api as is, as much as possible because I think this would be used wrapped from another source (like Airflow) where you would be able to map strings to concrete integers. I would create a separate command which would be used from an engineer. It could include the mappings you suggested, and other fancy things. Your request would break that.

Reply:
Source: https://github.com/dfinity/ic/blob/master/rs/nns/governance/src/gen/ic_nns_governance.pb.v1.rs#L2715

Reply:
> If you take a look at how ListProposalInfo struct is formatted it also expects a Vec<i32>. 

Correct, which means the tool has to take the list of strings specified, convert them to the right integers, and assemble the right `Vec` on behalf of the user.

The software exists to serve the needs of the people, not the other way around.

Reply:
For this command specifically. I think its best if we left it mapped exactly as is to the upstream with little to no mappings. "Software" part should come in a different command which will be more userfriendly and tailored to the need. This command should be there and integrated with different tools. If i add mapping here from strings to int you will have to add a mapping from int to string in Airflow (and you don't have that right now, you have enums) meaning we achieved nothing and have to maintain two maps in sync. This is just a map from the call of a canister allowed to the user via cli. Your request will be granted in a different command.

Reply:
OK.

Reply:
Can we use `strum` to map enums to strings and back?

Reply:
Probably yes, but it doesn't make sense because neither will the input from user be that enum, nur will the governance canister accept an enum

Reply:
Aha, there is already a [conversion to string](https://sourcegraph.com/github.com/dfinity/ic@09ce93457a49ba4ca452cf9e4d2d46dd54b3d146/-/blob/rs/nns/governance/src/gen/ic_nns_governance.pb.v1.rs?L3372-3390)
and [conversion from string](https://sourcegraph.com/github.com/dfinity/ic@09ce93457a49ba4ca452cf9e4d2d46dd54b3d146/-/blob/rs/nns/governance/src/gen/ic_nns_governance.pb.v1.rs?L3392-3411)